modules/ROOT/pages/client-id-based-policies.adoc

```diff
@@ -1,61 +1,51 @@
 
-In Mule 3.8.5 or later, the following OAuth 2.0 Token Enforcement policies include a client ID enforcement validation:
-
-* xref:external-oauth-2.0-token-validation-policy.adoc[Mule OAuth 2.0 Access Token Enforcement]
-* xref:policy-ping-federate.adoc[PingFederate OAuth 2.0 Token Enforcement]
-* xref:policy-openid-connect.adoc[OpenID Connect OAuth 2.0 Token Enforcement]
-
-= Offline mode
+Although token enforcement policies can optionally skip client validation, it is recommended that you enforce the validation to ensure that the token is associated to an approved contract.
 
-Credentials data from client application credentials registered to protected APIs, are provided by Anypoint Platform. Once a policy requiring such information is applied (eg Client ID enforcement policy), the Runtime caches persistently the information locally updating it periodically.
+== How the Policy Works
 
-Besides its performance benefits, this action enables the policy to work offline in the event of a sudden loss of connectivity to the platform. The enforcement is applied regardless of the state of the management plane, with existing data. Once connectivity is re-established, the cache will be periodically updated again.
+Before a client application is allowed to consume an API protected by a Client ID enforcement policy, the client must request access to the API. After an approved contract exists between the application and the API, every request must include the client credentials in compliance with how the policy is configured.
 
-= REST Console
-To prevent user requests from being rejected when using the REST Console, create a trait in the RAML root and then reference it in every operation of your API affected by policy. You may find the RAML snippet link containing the RAML code you need to add to the RAML in the corresponding policy from the list of applied policies in the Policies tab of your API specification in API Manager.
+For example, if the policy is configured to expect a client ID and client secret as headers, the application must send those credentials in the request, using the corresponding headers:
 
 image::raml-snippet.png[]
 
-= Credentials source
+== Configuring How the Policy Obtains Credentials
 
-The policy can be configured to extract the client ID and client secret from the HTTP request in different ways using DataWeave 2.0 expressions.
+//You can configure the Client ID Enforcement policy to obtain the client credentials in one of two different ways: by using an HTTP basic authentication header or by using a custom expression.
 
-The following examples assume that the values for application credentials are '1234' for client ID and 'abcd' for client secret.
+You can configure the policy to extract either both the client ID and client secret, or only the client ID from the HTTP request by using a variety of custom DataWeave 2.0 expressions. The following examples use a client ID of '1234' and a client secret of 'abcd'.
```

Comment:
Using DataWeave 2.0 is only valid in Mule 4.0.0+
For 3.x Mules customer should use MEL expressions.

Reply:
You further detail this later on, feel free to close this if you think it is not necessary to put it here too.

Reply:
Ok! Thanks!